### DISCUSSION TITLE
Nested block

### DISCUSSION THREAD

**Discussion by azar-maghsoodi** [area17/twill#2242] [Q&A]
Hello
In nested block, even when I define my block name in outer block, all the block is shown in outer block, I want to just a specific block be in the outer block, not all the blocks, Is it possible? is there any sample of nested block in Demo?

**Comment by haringsrob** [comment]
Hey @azar-maghsoodi in twill 3 you can have nested blocks, these are documented here:

https://twillcms.com/docs/block-editor/nested-blocks.html

There is currently an open pr however to fix an issue with them if you use component based blocks.

**Comment by azar-maghsoodi** [comment]
Hello @haringsrob 
Thank you for the response.
I already read the documnet and I mentioned the inner block like this <x-twill::block-editor name="quote"/>, but all the blocks that I am, are shown as a inner block and in addition a block namely body text is shown as a inner block witch is extra and is not my block, probably it is block from twill.

**Reply by haringsrob** [reply]
I am not sure I fully understand what you mean with inner block. Twill should not add any additional markup to your system.

Can you maybe share a repo that illustrates the issue, this will make it easier for us to debug and understand the issue.

**Reply by azar-maghsoodi** [reply]
> Hello @haringsrob Thank you for the response. I already read the documnet and I mentioned the inner block like this <x-twill::block-editor name="quote"/>, but all the blocks that I am, are shown as a inner block and in addition a block namely body text is shown as a inner block witch is extra and is not my block, probably it is block from twill.

For those who have this problem try to add 
'block_editor' => [
        'use_twill_blocks' => [],
    ]
    
    in the config/twill.php then you dont see twill default blocks (https://twillcms.com/guides/page-builder-with-blade/creating-a-block.html)

**Comment by azar-maghsoodi** [comment]
unfortunatly I dont have repo
But I try to explain more clear here.
I have a block, outer block namely landmark and the blade file is located in the resources/view/twill/blocks folder. and the code is

```
@twillBlockTitle('landmarks')
@twillBlockIcon('b-document') 
<x-twill::input
    name="title"
    label="Title"
    :maxLenght="100"
/>

<x-twill::block-editor name="quote"/> 
```

and quote block in my inner block and I want to show just my quote block in the my landmark block.

and my quote block blade is in the folder  resources/view/twill/blocks and the below is it`s code:
```
@twillBlockTitle('Text')
@twillBlockIcon('b-quote') 
@php
$wysiwygOptions = [
    ['header' => [2, 3, 4, 5, 6, false]],
    'bold',
    'italic',
    'underline',
    'strike',
    'blockquote',
    "code-block",
    'ordered',
    'bullet',
    'hr',
    'code',
    'link',
    'clean',
    'table',
    
];
@endphp
<x-twill::wysiwyg
    name="quote"
    :label="twillTrans('Text')"
    :toolbar-options="$wysiwygOptions"
    :placeholder="twillTrans('Text')"
    :edit-source="true"
    :maxlength="200"
    :note="twillTrans('Help text')"
/> .
```
Now in my outer block(landmark), i have all my block in my code plus a block namely Body Text.
I hope now be more clear.

**Reply by haringsrob** [reply]
I think you may want to use the "blocks" to limit which blocks can be added? 

https://twillcms.com/docs/form-fields/block-editor.html
